tools/accuracy_checker/accuracy_checker/evaluators/custom_evaluators/custom_text_recognition_evaluator.py

```diff
@@ -264,13 +274,78 @@ def __init__(self, network_info, launcher, models_args, meta, is_blob=None):
         })
         if not contains_all(network_info, ['recognizer_encoder', 'recognizer_decoder']):
             raise ConfigError('network_info should contain encoder and decoder fields')
-        self.vocab = meta['vocab']
         self.recognizer_encoder = create_recognizer(network_info['recognizer_encoder'], launcher, 'encoder')
         self.recognizer_decoder = create_recognizer(network_info['recognizer_decoder'], launcher, 'decoder')
         self.sos_index = 0
         self.eos_index = 2
         self.max_seq_len = int(network_info['max_seq_len'])
 
+    def get_phrase(self, indices):
+        raise NotImplementedError()
+
+    def predict(self, identifiers, input_data):
+        raise NotImplementedError()
+
+    def reset(self):
+        pass
+
+    def release(self):
+        self.recognizer_encoder.release()
+        self.recognizer_decoder.release()
+
+
+class SequentialTextRecognitionModel(BaseSequentialModel):
+    def __init__(self, network_info, launcher, models_args, meta, is_blob=None):
+        super().__init__(network_info, launcher, models_args, meta, is_blob=is_blob)
+        self.vocab = network_info['custom_label_map']
+
+    def get_phrase(self, indices):
+        res = ''
+        for idx in indices:
+            if idx != self.eos_index:
```

Comment:
do we really need this check?
as I can see idx == eos already filtered in predict method.

Reply:
The filtering was applied after adding logit to result tensor, changed this.